File: java/producer/src/main/java/br/com/dynamiclight/push/PushController.java
package br.com.dynamiclight.push;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("v1")
public class PushController {

    private final PushService pushService;

    public PushController(PushService pushService) {
        this.pushService = pushService;
    }

    @PostMapping("/push/send")
    private ResponseEntity<?> send(@RequestBody Push message) {
        pushService.send(message);
        return ResponseEntity.accepted().build();
    }

}

File: kotlin/producer/src/main/kotlin/br/com/dynamiclight/push/PushController.kt
package br.com.dynamiclight.push

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("v1")
class PushController(val pushService: PushService) {

    @PostMapping("/push/send")
    private fun send(@RequestBody message: Push) : ResponseEntity<*> {
        pushService.send(message)
        return ResponseEntity.accepted().build<ResponseEntity<*>>()
    }

}